src/ProjectNeon/Assets/Features/Battle/Targeting/BattleStateTargetingExtensions.cs

```diff
@@ -2,27 +2,44 @@
 
 public static class BattleStateTargetingExtensions
 {
-    // @todo #1:30min Needs to handle Group All scenarios.
-
     private static Member[] Get(this BattleState state, TeamType team)
     {
         return state.Members.Values.Where(x => x.TeamType == team).ToArray();
     }
 
     public static Target[] GetPossibleTargets(this BattleState state, Member self, Group group, Scope scope)
-        => group == Group.Self
-            ? new Target[] { new MemberAsTarget(self) }
-            : NonSelfTargetsFor(state, self.TeamType, group, scope);
+    {
+        Target[] targets = new Target[0];
+        switch (group)
+        {
+            case Group.Self:  {
+                targets = new Target[] { new MemberAsTarget(self) };
+                break;
+            }
+            case Group.All:  {
+                targets = Targets(new Team(state.Members.Values));
```

Comment:
Enemies and allies aren't on the same team. Maybe we need one new construct for a target that contains members from multiple teams.

Reply:
@SilasReinagel Maybe `Team` isn't the right name for this construct. `Team` here is just a synonym for a group of members, if they are hero or enemies, or both, or environmental elements, it just doesn't matter, specially because we can have effects that could affect all the combatants (meteor strike, exploding scenery barrel, etc), or just part or any combination of them (Will we be able to target only half of enemies with an area attack (crhono trigger style)? Will we be able to target only two members of our party with a heal spell (as a priest which heals other but does not heal self)? ). We don't use it as any other thing than a `Member` collection, so, conceptually its name may be wrong but its function is okay, regardless the fact that enemies, heroes and environmental props aren't on the same team.

Maybe we should call it `Members`? But hey, what's the purpose of this object? If we aren't using the `Composite` pattern in the `Target`-`Member`-`Team` relation, i.e., if `Team` and `Member` are just one item and one collection of that item, I think that we could strip `Team` from the project and get along with a `Member` array, plain and simple.

So, what will it be? Using the composite pattern will allow us to apply card effects in a object- oriented way, delegating to the `Leaf` (`Member`) and `Composite` (`Team` ) elements of the pattern the effect-dealing (making each `Member` know how to deal with some specific effect), but using just the `Member` array for everything would spare us of a more complex design. Anyways, `Team` name is inadequate and must be changed either the solution we adopt.

Reply:
@paulodamaso It seems like it would be fine just to have a first-class collection. Maybe `Team` and `Target` don't need a `TeamType`. `Members` is probably fine.